### PR TITLE
simplified pg_rect_clip

### PR DESCRIPTION
**Pull request by itzpr3d4t0r**
As the title suggests.

### CODE REVIEW

**Comment by hpnok** [comment]
Could you inline the `goto` at this point?

**Comment by itzpr3d4t0r** [comment]
> Could you inline the `goto` at this point?
Implemented, thanks.

File: src_c/rect.c [hunk 1]
```diff
@@ -1107,46 +1107,16 @@ pg_rect_clip(pgRectObject *self, PyObject *args)
...
-        goto nointersect;
+    x = MAX(A->x, B->x);
+    y = MAX(A->y, B->y);
+    w = MIN(A->x + A->w, B->x + B->w) - x;
+    h = MIN(A->y + A->h, B->y + B->h) - y;
 
-    /* Top */
-    if ((A->y >= B->y) && (A->y < (B->y + B->h))) {
-        y = A->y;
+    if (w < 0 || h < 0) {
```
Comment (dr0id) [inline src_c/rect.c:1115]: I think this should be 
`if(w <= 0 || h <= 0) {`

See other comment for detailed explanation.

Comment (dr0id) [reply src_c/rect.c:1115]: I miss some unittests (haven't checked if there are any). Base on [this overview of rect overlapping combinations](http://gandraxa.com/detect_overlapping_subrectangles.xml#B10) I wrote [this script](https://pastebin.com/3hbTru07) to check your changes.

Comment (itzpr3d4t0r) [reply src_c/rect.c]: thanks, yeah i noticed that tests don't cover all cases, i should add some more

**Review by MyreMylar — APPROVED** [review]
LGTM 👍 

Test program used:

```
import random

import pygame

pygame.init()

rect_1 = pygame.Rect((275, 150), (250, 250))
rect_2 = pygame.Rect((350, 240), (200, 200))


window_surf = pygame.display.set_mode((800, 600))

background = pygame.Surface((800, 600))
background.fill((255, 255, 255))

running = True

while running:

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False

    window_surf.blit(background, (0, 0))

    rect_1.center = pygame.mouse.get_pos()

    clip_rect = rect_1.clip(rect_2)

    pygame.draw.rect(window_surf, pygame.Color('#FF0000'), rect_1, 1)
    pygame.draw.rect(window_surf, pygame.Color('#00FF00'), rect_2, 1)
    pygame.draw.rect(window_surf, pygame.Color('#0000FF'), clip_rect, 1)

    pygame.display.update()

```

**Review by ankith26 — APPROVED** [review]
LGTM! The code looks much simpler now, thanks for the PR :sunglasses:

**Pull request merged**